code/modules/mob/living/simple_animal/hostile/space_dragon.dm

```diff
@@ -0,0 +1,727 @@
+/// The carp rift is currently charging.
+#define CHARGE_ONGOING			0
+/// The carp rift is currently charging and has output a final warning.
+#define CHARGE_FINALWARNING		1
+/// The carp rift is now fully charged.
+#define CHARGE_COMPLETED		2
+/// The darkness threshold for space dragon when choosing a color
+#define DARKNESS_THRESHOLD		50
+
+/**
+ * # Space Dragon
+ *
+ * A space-faring leviathan-esque monster which breathes fire and summons carp.  Spawned during its respective midround antagonist event.
+ *
+ * A space-faring monstrosity who has the ability to breathe dangerous fire breath and uses its powerful wings to knock foes away.
+ * Normally spawned as an antagonist during the Space Dragon event, Space Dragon's main goal is to open three rifts from which to pull a great tide of carp onto the station.
+ * Space Dragon can summon only one rift at a time, and can do so anywhere a blob is allowed to spawn.  In order to trigger his victory condition, Space Dragon must summon and defend three rifts while they charge.
+ * Space Dragon, when spawned, has five minutes to summon the first rift.  Failing to do so will cause Space Dragon to return from whence he came.
+ * When the rift spawns, ghosts can interact with it to spawn in as space carp to help complete the mission.  One carp is granted when the rift is first summoned, with an extra one every 30 seconds.
+ * Once the victory condition is met, all current rifts become invulnerable to damage, are allowed to spawn infinite sentient space carp, and Space Dragon gets unlimited rage.
+ * Alternatively, if the shuttle arrives while Space Dragon is still active, their victory condition will automatically be met and all the rifts will immediately become fully charged.
+ * If a charging rift is destroyed, Space Dragon will be incredibly slowed, and the endlag on his gust attack is greatly increased on each use.
+ * Space Dragon has the following abilities to assist him with his objective:
+ * - Can shoot fire in straight line, dealing 30 burn damage and setting those suseptible on fire.
+ * - Can use his wings to temporarily stun and knock back any nearby mobs.  This attack has no cooldown, but instead has endlag after the attack where Space Dragon cannot act.  This endlag's time decreases over time, but is added to every time he uses the move.
+ * - Can swallow mob corpses to heal for half their max health.  Any corpses swallowed are stored within him, and will be regurgitated on death.
+ * - Can tear through any type of wall.  This takes 4 seconds for most walls, and 12 seconds for reinforced walls.
+ */
+/mob/living/simple_animal/hostile/space_dragon
```

Comment:
Delete the rifts on destroy

Reply:
Removed on death **if** their objective is not complete. Rifts are supposed to stay if they succeed and then die so this would break that.